### DISCUSSION TITLE
Running on mac

### DISCUSSION THREAD

**Discussion by bbaraniec** [christgau/wsdd#103] [Q&A]
Not an issue per say. Just wanted to ask if anyone managed to run this on macOS as service.
Always when run in plist, app get terminated with code error 2. Running manually, as bash script works.

**Comment by christgau** [comment]
Oh, wow, wsdd runs on macOS. Thanks for that information.

Since I don't have access to an macOS installation, I can only guess. Error 2 is likely to be ENOENT, i.e. "no such file or directory". That's what it is on Linux and on FreeBSD, and because macOS is an BSD under the hood as well, I'd guess the error code has the same meaning.

Could it be that the service definition in the plist file (?) tries to start wsdd as in the shell, i.e. using the path only (`/path/to/wsdd.py`). Maybe it helps to [specify the Python interpreter](https://stackoverflow.com/questions/36725728/how-to-run-a-python-command-in-a-plist): `/usr/bin/python3 /path/to/wsdd.py`. During system/service startup `usr/bin/env` may not find the interpreter due to the settings of PATH and therefore the script won't start.

**Comment by bbaraniec** [comment]
I have tried to define whole path to python3, I have tried to execute the .sh script that works, when I start it manually, or from automator. I have tried to run it from crontab, always same error.
I love the option to have it running, but spinning wheel on top bar drives me nuts :)
I wish log would say more. Not giving up just yet.

**Comment by christgau** [comment]
Not sure which ".sh script" you mean. However, since I don't have access to a Mac - as I said - I'm affraid that I can't help you much more than what I've wrote. Maybe someone else can help you. If you find a solution, please share with the community here so that others can benefit from it as well. If the service itself works fine and wsdd also runs as expected, you may contribute the plist file.

**Comment by bbaraniec** [comment]
I meant that I wrote a simple bash script tp start it. Anyway, just wanted to see if anyone was indeed using it on mac. Maybe at some point you will be bored enough to install mac on vmplayer :)
